src/resolvers/helpers/filter.d.ts

```diff
@@ -2,21 +2,18 @@ import { ObjectTypeComposerArgumentConfigMapDefinition, ObjectTypeComposer } fro
 import { Model } from 'mongoose';
 import { MongoId } from '../../types/mongoid';
 import { ExtendedResolveParams } from '../index';
-import { FilterOperatorsArgs, FilterOperatorsOpts } from './filterOperators';
+import { FilterOperatorsArgs } from './filterOperators';
+
 
 export type FilterHelperArgsOpts = {
   filterTypeName?: string;
   isRequired?: boolean;
   onlyIndexed?: boolean;
-  requiredFields?: string | string[];
```

Comment:
Let the `schema` dictate whether this is required

Reply:
Nope, it's a bad decision.

We should allow to user to override required fields from schema via customizationOptions –  https://github.com/graphql-compose/graphql-compose-mongoose#customization-options

For example, the schema has 5 optional fields, and for API consumers decided to do that several fields should be required. Via custOpts it much easy to do rather than traversing types via `graphql-compose` methods and modify types manually.

Reply:
Sure, but we'd need a tree to handle configuration. For `operators`, with this PR you can do this now:

```
export const UsersComposition = composeWithMongoose(Users, {
  findResolverName: 'findMany',
  resolvers: {
    findMany: {
      filter: {
        operators: {
          payment: ['exists'],
          billingAddress: {
            state: ['in']
          }
        }
      }
    }
  }
```

`isRequired` is just a flag to say 'respect the mongoose model', we could have another tree for `requiredFields` and `removeFields`. But I think we need a better plan for that since rebuilding those trees feels like duplicating what mongoose does itself, I totally get the reasoning behind `requiredFields` and `removeFields`, so I guess my question is, we should support trees for these? (yes, imo). I just want to kick the bucket and get nesting support in.

Reply:
`isRequired` is just a flag to ~say respect the mongoose model~ make this arg required in graphql schema

`operators: { payment: ['exists'], ...` – i totally love this thing 👍

`we should support trees for these?` – yes we should somehow. I suggest looking at TypePolicies which introduced in ApolloClient 3 https://www.apollographql.com/docs/react/caching/cache-configuration/#typepolicy-fields I think we can borrow their design and make something similar with an introduction of some functions to config (e.g. before/after hooks).

So with publishing MAJOR version we are totally free to reimplement customization options from scratch.